lib/openstack_query/docs/QUERIES.md

```diff
@@ -0,0 +1,232 @@
+           meta_param_2 = None,
+           ...
+       ):
+           # Define logic here that will alter the keyword arguments that will be passed to the openstacksdk command
+           # based on the meta-parameter values passed in. This method should return a dictionary which will be merged with
+           # server-side filters and be used when calling the specific openstacksdk query which gets the resources being queried for
+           ...
+
+       def _run_query(
+           self,
+           conn: OpenstackConnection,
+           filter_kwargs: Optional[ServerSideFilters] = None,
+           **meta_params,
+       )
+           # Define logic here that will setup keyword arguments - merging server-side kwargs (filter_kwargs) and meta_param
+           # and call run_paginated_query
+           ...
+
+           # here we call `run_paginated_query` with the "dummy" openstacksdk call "conn.compute.resource"
+           # and we call it with key-word arguments, merging both filter_kwargs with parsed meta-params returned from
+           # self._parse_meta_params
+
+           # see RUNNER.md for more info on paginated queries
+
+           # you can other types of queries/logic here - this is up to your discretion
+           return self._run_paginated_query(conn.compute.resource, **{filter_kwargs, meta_params})
+   ```
+
+#### 3. Create a Mapping Class
+
+A new subclass of the `MappingInterface` class is required to store query mappings (see above).
+
+To add a Mapping Class:
+1. Create a new file under `/openstack_query/mappings/` called `<resource>_mapping.py` replace `<resource>` with openstack resource name you want to query
+2.  Create the `<resource>Mapping` class definition
+   - class must inherit from `MappingInterface` abstract base class and implement abstract methods.
+   - see example below
+```python
+
+class ResourceMapping(MappingInterface):
+    ...
+
+    @staticmethod
+    def get_chain_mappings():
+        # here you need to define if there any shared-properties between this query's properties and other queries.
+        # used when chaining. See CHAINING.md for more info
+
+    @staticmethod
+    def get_runner_mapping() -> Type[ResourceRunner]:
+        """
+        Returns a mapping to associated Runner class for the Query (ResourceRunner)
+        """
+        # this is using the "dummy" ResourceRunner defined earlier
+        return ResourceRunner
+
+    @staticmethod
+    def get_prop_mapping() -> Type[FlavorProperties]:
+        """
+        Returns a mapping of valid presets for server side attributes (ResourceProperties)
+        """
+        # this is using the "dummy" ResourceProperties defined earlier
+        return ResouceProperties
+
+    @staticmethod
+    def get_server_side_handler() -> ServerSideHandler:
+       ...
+        # here you need to define server-side filters. See PRESETS.md for details
+
+    @staticmethod
+    def get_client_side_handlers() -> ServerSideHandler:
+       ...
```

Comment:
Could you add docstrings to these static methods like the methods above?

Reply:
These aren't actual methods - its pseudo-code - basically instructions of methods you need to edit. 

I don't want people to copy-paste without reading so I didn't copy the entire method

Reply:
Ahh I see, okay 👍